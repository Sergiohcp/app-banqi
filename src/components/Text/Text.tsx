import React from "react";
import { View, Text as ReactText, TextProps } from "react-native";

import { colors } from "@styles/colors";
import { fonts } from "@styles/fonts";

interface IText {
  fontWeight?: "regular" | "bold";
  fontSize?: number;
  color?: string;
  textStyle?: TextProps;
}

export const Text: React.FC<IText> = ({
  children,
  fontWeight = "regular",
  fontSize = fonts.size.fontSize12,
  color = "black",
  textStyle,
  ...rest
}) => {
  const fontWeights = {
    regular: fonts.family.regular,
    bold: fonts.family.bold,
  };

  function getTextColor() {
    switch (color) {
      case "pink":
        return colors.pink;

      case "cyan":
        return colors.cyan;

      case "green":
        return colors.green;

      case "black":
        return colors.black;

      default:
        return color;
    }
  }

  function getStyle() {
    const style = {
      fontFamily: fontWeights["regular"],
      fontSize,
      color: getTextColor(),
    };

    if (!!fontWeights[fontWeight]) {
      style.fontFamily = fontWeights[fontWeight];
    }

    if (!!fontSize) {
      style.fontSize = fontSize;
    }

    return style;
  }

  return (
    <View {...rest}>
      <ReactText style={[getStyle(), textStyle]}>{children}</ReactText>
    </View>
  );
};
