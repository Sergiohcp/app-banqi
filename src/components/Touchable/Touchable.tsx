import React from "react";
import { TouchableOpacity, TouchableWithoutFeedback, View } from "react-native";

interface ITouchable {
  type?: "Opacity" | "WithoutFeedback";
  onPress: () => void;
}

const Touchable: React.FC<ITouchable> = ({
  children,
  type = "Opacity",
  onPress,
  ...rest
}) => {
  function getTouchableType() {
    switch (type) {
      case "WithoutFeedback":
        return (
          <TouchableWithoutFeedback onPress={onPress}>
            <View>{children}</View>
          </TouchableWithoutFeedback>
        );

      default:
        return (
          <TouchableOpacity onPress={onPress} {...rest}>
            {children}
          </TouchableOpacity>
        );
    }
  }

  return getTouchableType();
};

export default Touchable;
