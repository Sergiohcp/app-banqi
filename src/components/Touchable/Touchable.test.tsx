import { Text } from "@components/Text";
import React from "react";
import renderer from "react-test-renderer";

import Touchable from "./Touchable";

describe("Touchable test", () => {
  it("should take snapshot of component", () => {
    const tree = renderer
      .create(
        <Touchable onPress={() => {}}>
          <Text>Snapshot Test</Text>
        </Touchable>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
