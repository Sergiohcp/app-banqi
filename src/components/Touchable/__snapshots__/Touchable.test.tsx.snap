// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable test should take snapshot of component 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": undefined,
          "alignSelf": "auto",
          "backgroundColor": "transparent",
          "borderRadius": undefined,
          "flex": 0,
          "flexDirection": "column",
          "height": undefined,
          "justifyContent": "flex-start",
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 12,
            "opacity": undefined,
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Snapshot Test
    </Text>
  </View>
</View>
`;
